dropwizard/src/main/java/com/linecorp/armeria/server/dropwizard/ArmeriaServerFactory.java

```diff
@@ -0,0 +1,311 @@
+            sb.accessLogWriter(AccessLogWriter.disabled(), true);
+        }
+        return sb;
+    }
+
+    @JsonProperty
+    private @Valid ConnectorFactory connector = ArmeriaHttpConnectorFactory.build();
+    @JsonProperty
+    private @Valid @NotNull AccessLogWriterFactory accessLogWriter = new CommonAccessLogWriterFactory();
+    @JsonProperty
+    private boolean jerseyEnabled = true;
+    @JsonProperty
+    private @MinSize(0) Size maxRequestLength = Size.bytes(Flags.defaultMaxRequestLength());
+    @JsonProperty
+    private boolean disableDateHeader;
+    @JsonProperty
+    private boolean disableServerHeader;
+    @JsonProperty
+    private boolean verboseResponses;
+    @JsonProperty
+    @Nullable
+    private String defaultHostname;
+    @JsonIgnore
+    private transient ServerBuilder serverBuilder;
+
+    public ArmeriaServerFactory() {
+    }
+
+    /**
+     * Sets up the Armeria ServerBuilder with values from the Dropwizard Configuration.
+     * Ref <a href="https://line.github.io/armeria/advanced-production-checklist.html">Production Checklist</a>
+     *
+     * @param serverBuilder A non-production ready ServerBuilder
+     * @return A production-ready ServerBuilder
+     */
+    // visible for testing
+    ServerBuilder decorateServerBuilderFromConfig(final ServerBuilder serverBuilder) {
+        Objects.requireNonNull(serverBuilder);
+        serverBuilder.maxNumConnections(getMaxQueuedRequests());
+        serverBuilder.blockingTaskExecutor(Executors.newFixedThreadPool(getMaxThreads()), true);
+        serverBuilder.maxRequestLength(maxRequestLength.toBytes());
+        serverBuilder.idleTimeoutMillis(getIdleThreadTimeout().toMilliseconds());
+        serverBuilder.gracefulShutdownTimeout(
+                Duration.ofSeconds(30L),
+                Duration.ofMillis(getShutdownGracePeriod().toMilliseconds()));
+        if (isDateHeaderDisabled()) {
+            serverBuilder.disableDateHeader();
+        }
+        if (isServerHeaderDisabled()) {
+            serverBuilder.disableServerHeader();
+        }
+        serverBuilder.verboseResponses(hasVerboseResponses());
+        Optional.ofNullable(getDefaultHostname()).ifPresent(serverBuilder::defaultHostname);
+        // TODO: Add more items to server builder via Configuration
+        return serverBuilder;
+    }
+
+    private String getDefaultHostname() {
+        return defaultHostname;
+    }
+
+    @JsonGetter("verboseResponses")
+    private boolean hasVerboseResponses() {
+        return verboseResponses;
+    }
+
+    private void setVerboseResponses(boolean verboseResponses) {
+        this.verboseResponses = verboseResponses;
+    }
+
+    @JsonGetter("disableServerHeader")
+    private boolean isServerHeaderDisabled() {
+        return disableServerHeader;
+    }
+
+    @JsonGetter("disableDateHeader")
+    public boolean isDateHeaderDisabled() {
+        return disableDateHeader;
+    }
```

Comment:
How about renaming to `isServerHeaderEnabled()` and `isDateHeaderEnabled()` to avoid double negation which may be confusing? The default value could be `true` for both properties.